articles/active-directory/develop/quickstart-v2-java-webapp.md

```diff
@@ -33,86 +32,124 @@ When you've completed the guide, your application will accept sign-ins of person
+attributes.
 
 #### Step 2: Download the code sample
- 
+
  [Download the Code Sample](https://github.com/Azure-Samples/ms-identity-java-webapp/archive/master.zip)
- 
- #### Step 3: Configure the code sample 
- 
+
+#### Step 3: Configure the code sample
+
  1. Extract the zip file to a local folder.
  1. If you use an integrated development environment, open the sample in your favorite IDE (optional).
  1. Open the **application.properties** file, which can be found in *src/main/resources/*.
  1. Replace application properties.
-   1. Find `aad.clientId` and update the value of `Enter_the_Application_Id_here` with the **Application (client) ID** value of the application you registered. 
-   1. Find `aad.authority` and update the value of `Enter_the_Tenant_Name_Here` with the **Directory (tenant) ID** value of the application you registered.
-   1. Find `aad.secretKey` and update the value of `Enter_the_Client_Secret_Here` with the **Client Secret** you created in **Certificates & Secrets** for the application you registered.
+    1. Find `aad.clientId` and update the value of `Enter_the_Application_Id_here` with the **Application (client) ID** value
+    of the application you registered.
+    1. Find `aad.authority` and update the value of `Enter_the_Tenant_Name_Here` with the **Directory (tenant) ID** value of
+    the application you registered.
+    1. Find `aad.secretKey` and update the value of `Enter_the_Client_Secret_Here` with the **Client Secret** you created in
+    **Certificates & Secrets** for the application you registered.
 
 > [!div renderon="docs"]
 > Where:
 >
 > - `Enter_the_Application_Id_here` - is the Application Id for the application you registered.
-> - `Enter_the_Client_Secret_Here` - is the **Client Secret** you created in **Certificates & Secrets** for the application you registered.
+> - `Enter_the_Client_Secret_Here` - is the **Client Secret** you created in **Certificates & Secrets** for the application
+you registered.
 
 #### Step 4: Run the code sample
-1. Run the code sample, and open a browser and navigate to *http://localhost:8080*.
-1. The front page contains a **sign-in** button. Click on the **sign-in** button to redirect to Azure Active Directory. The user will be prompted for their credentials.  
-1. After successfully authenticating on Azure Active Directory, they will be redirected to *http://localhost:8080/msal4jsamples/secure/aad*. They are officially signed in to the application, and the page should show information for the signed in account. It will also contain buttons for: 
+
+To run the project, you can either:
+
+- Run it directly from your IDE by using the embedded spring boot server
+- or package it to a WAR file using [maven](https://maven.apache.org/plugins/maven-war-plugin/usage.html) and deploy it to a J2EE container solution such as [Apache Tomcat](https://tomcat.apache.org/maven-plugin-trunk/tomcat6-maven-plugin/examples/deployment.html).
```

Comment:
remove 'or' and put 'Package it to a WAR ...'

Reply:
I think this means the way of showing "either","or" (either run from IDE or package it to WAR)